spec/lib/xdrgen/rust_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'spec_helper'
+
+describe Xdrgen::Generators::Rust do
+  generator_fixture_paths.each do |path|
+    it "can generate #{File.basename path}" do
+      c = generate path, ""
+    end
+
+    it "can generate #{File.basename path} with custom str impls" do
+      c = generate path, "_custom_str_impls", {
+        rust_types_custom_str_impl: [
+          "Foo",
```

Comment:
What are these types and what is this file doing?

Reply:
This tests loads xdr types from the fixtures directory, and generates code for them. I'm picking a handful of types, one for each type of xdr type (struct, enum, etc), and marking them as having custom str implementations. The .rs files being committed in the test directory show the generated code, and that no str implementations have been included for those types.